src/main/java/hudson/plugins/ircbot/v2/IRCColorizer.java

```diff
@@ -40,25 +175,108 @@ public static String colorize(String message){
         }
     }
     
+    /**
+     * Colorize the message line if certain keywords are found in it. 
+     * @param nickname user's nickname
+     * @param message jenkin's message
+     */
+    public static String colorize(String nickname, String message) {
+    	if(userPattern.containsKey(nickname)){
+            return user_colorize(nickname, message);
+    	} else {
+    		return colorize(message);
+    	}
+    }
+    
+    /**
+     * set the user defined color for specific string,
+     * will be called by onMessage() method in PircListener.java
+     * @param nickname user's nickname
+     * @param pattern patterns that user wants to change, it also includes regex
+     * @param color e.g RED, BLUE, GREEN
+     */
+    public static void setter(String nickname, String pattern, String color) 
+    {
+        // get hashmap for userid or create a new one
+        HashMap<String, String> user_hash;
+        if(userPattern.containsKey(nickname)){
+            user_hash = userPattern.get(nickname);
+        } else {
+            user_hash = new HashMap<String, String>();
+        }
+        
+        for(String name: user_hash.keySet()) {
+        	System.out.println(name + " : " + user_hash.get(name));
+        }
+
+        // with user_hash add Pattern and String-color
+        user_hash.put(pattern, color);
+        
+        // put user_hash back into userPattern
+        userPattern.put(nickname, user_hash);
+        
+        // Serialize userPattern to disk
+        writeFile("userPattern.ser");        
+    }
+    
+    // creates default Themes
+    private static void createThemes() {
+    	themes = new HashMap<String, HashMap<ResultTrend,String>>();
+    	themes.put("THEME1", createDefault());
+    	themes.put("THEME2", oneColorTheme("CYAN"));
+    	themes.put("THEME3", theme2());
```

Comment:
could we have more descriptive names for the themes?
NB: put("THEME3", theme2()); !?